pandas/core/strings.py

```diff
@@ -323,15 +328,23 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     n : int, default -1 (all)
         Number of replacements to make from start
-    case : boolean, default True
-        If True, case sensitive
-    flags : int, default 0 (no flags)
-        re module flags, e.g. re.IGNORECASE
+    case : boolean, optional
+        - if False, case insensitive
+        - Must be None if `pat` is a compiled regex
```

Comment:
So I think we need to have ``case=None``. Then you say in the doc-string (you have 2 lines)

- if pat is compiled expression then ``case`` must be ``None``
- if pat is a string, then case defaults to ``True``.

I don't necessarily like this API, but not sure what else we can do.